code/modules/mob/mob_movement.dm

```diff
@@ -500,7 +500,7 @@
 					if(observer.invisibility == 0 || observer.mind && (find_active_faction_by_member(observer.mind.GetRole(LEGACY_CULTIST)) || find_active_faction_by_member(iscultist(observer))))
 						to_chat(mob, "<span class='warning'>You cannot get past holy grounds while you are in this plane of existence!</span>")
 						blocked = TRUE
-				else if(!blocked && !Move_object(direct) && !Dir_object(direct))
+				if(!blocked && !Move_object(direct) && !Dir_object(direct))
```

Comment:
This implementation will spam ghosts with how they can't get past holy grounds

Reply:
it already did that before